### PR TITLE
Curl improvements

### PR DESCRIPTION
**Pull request by h9k**
enforce verifying ssl peer certificate

default behaviour in curl

### CODE REVIEW

**Comment by h9k** [comment]
I made the changes you proposed and some other small things, however I have difficulties testing (authorization failed).
As I don't have any more time for this morning, I will have to check this evening what is going on.
That's why I marked this PR as WIP for now.

**Comment by h9k** [comment]
Ok, the problem was the POST method, which RRPproxy seems not to like, so I reverted to GET. Works now!

File: modules/registrars/rrpproxy/rrpproxy.php [hunk 1]
```diff
@@ -26,6 +26,8 @@
 use WHMCS\Domains\DomainLookup\SearchResult;
 use WHMCS\Module\Registrar\RRPProxy\RRPProxyClient;
 
+define(RRPPROXY_VERSION, "0.1.0");
```
Comment (papakai) [inline modules/registrars/rrpproxy/rrpproxy.php:29]: super :-)

File: modules/registrars/rrpproxy/lib/RRPProxyClient.php [hunk 2]
```diff
@@ -58,18 +58,26 @@ public function call($command, $params = array())
...
+        $ch = curl_init($this->api_url);
+        curl_setopt_array($ch, [
+            CURLOPT_TIMEOUT         => 20,
+            CURLOPT_COOKIESESSION   => true,
+            CURLOPT_SSL_VERIFYPEER  => true,
+            CURLOPT_RETURNTRANSFER  => true,
+            CURLOPT_POST            => true,
+            CURLOPT_POSTFIELDS      => $params,
+            CURLOPT_HEADER          => false,
+            CURLOPT_USERAGENT       => "WHMCS (" . PHP_OS . "; " . php_uname('m') . "; rv:rrpproxy/".RRPPROXY_VERSION.") php/" . PHP_MAJOR_VERSION . "." . PHP_MINOR_VERSION . "." . PHP_RELEASE_VERSION,
```
Comment (papakai) [inline modules/registrars/rrpproxy/lib/RRPProxyClient.php:76]: `implode(".", [PHP_MAJOR_VERSION, PHP_MINOR_VERSION, PHP_RELEASE_VERSION])`

is probably better readable.

File: modules/registrars/rrpproxy/lib/RRPProxyClient.php [hunk 3]
```diff
@@ -36,7 +36,7 @@ public function __construct()
             $params[$data->setting] = self::decrypt($data->value);
         }
         if ($params['TestMode']) {
-            $this->api_url = 'https://api-ote.rrpproxy.net/api/call?s_opmode=OTE&s_login=' . urlencode($params['Username']) . '&s_pw=' . urlencode($params['Password']);
+            $this->api_url = 'https://api-ote.rrpproxy.net/api/call?s_opmode=OTE&s_login=' . urlencode($params['Username']) . '&s_pw=' . urlencode($params['TestPassword']);
         } else {
             $this->api_url = 'https://api.rrpproxy.net/api/call?s_login=' . urlencode($params['Username']) . '&s_pw=' . urlencode($params['Password']);
```
Comment (papakai) [inline modules/registrars/rrpproxy/lib/RRPProxyClient.php:41]: here, you use urlencode for parameter value line 66, you removed this.
At HEXONET, we use `rawurlencode` for both: key and value.
I suggest to follow that way (of course to be tested). rawurlencode is following RFC3986 which is good.
Check: https://stackoverflow.com/questions/996139/urlencode-vs-rawurlencode

cheers!

fyi: This urlencoding RFC topic is a big thing coming with differences from programming language to programming language. Just that you heard about this :-)

**Pull request merged**